### DISCUSSION TITLE
LLMs support 🤖

### DISCUSSION THREAD

**Discussion by assafelovic** [assafelovic/gpt-researcher#544] [Announcements]
Excited for yet another HUGE release that includes LLM support for the following: Llama3, Mistral, Anthropic, HuggingFace, Together AI, Gemini and more!

We've also updated our docs including examples:
- Customizing the research assistant: https://docs.gptr.dev/docs/gpt-researcher/config
- Configuring LLMs: https://docs.gptr.dev/docs/gpt-researcher/llms

This release also includes LangGraph deployment by @hwchase17 and additional stability improvements. Thank you to all the amazing contributions!

## What's Changed
* remove DOC_Path from docker-compose - rely on .env instead by @ElishaKay in https://github.com/assafelovic/gpt-researcher/pull/509
* Adding one-click deploy button for RepoCloud.io to README.md by @cosark in https://github.com/assafelovic/gpt-researcher/pull/515
* Fix Pydantic validation of base_url assignment for ChatOpenAI model by @mmashnev in https://github.com/assafelovic/gpt-researcher/pull/512
* Added Instructions for Groq by @dphiggs01 in https://github.com/assafelovic/gpt-researcher/pull/536
* Implemented GroqProvider by @dphiggs01 in https://github.com/assafelovic/gpt-researcher/pull/526
* add support for custom openai api embeddings  by @sebaxzero in https://github.com/assafelovic/gpt-researcher/pull/528
* langgraph deploy by @hwchase17 in https://github.com/assafelovic/gpt-researcher/pull/537
* Add logo to README by @assafelovic in https://github.com/assafelovic/gpt-researcher/pull/540
* added pandas dependency for reading csv, commented testing dependenci… by @ElishaKay in https://github.com/assafelovic/gpt-researcher/pull/541
* Adding support for Ollama (both LLM and embeddings) by @gschmutz in https://github.com/assafelovic/gpt-researcher/pull/527

## New Contributors
* @cosark made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/515
* @mmashnev made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/512
* @sebaxzero made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/528
* @gschmutz made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/527

**Full Changelog**: https://github.com/assafelovic/gpt-researcher/compare/v0.2.3...v0.2.4

<hr /><em>This discussion was created from the release <a href='https://github.com/assafelovic/gpt-researcher/releases/tag/v0.2.4'>LLMs support 🤖</a>.</em>

**Comment by Gallisatyricon** [comment]
Hi, I'm testing phi3 with Ollama in local. I have to say the results are pretty good on my old computer. Not as fast as Gpt4o, but to the point except for "Ressource report" that has some odd results. Need futher testing to see the patern. 
I tried phi3, because Llama3 is very slow and only working for web search (not at all on my documents). My computer knowledge is too weak to understand why. 
All that to say, thank you for this work that looks very promising! 👏
